AppCenter/AppCenter/Internals/Model/Util/MSAppCenterUserDefaults.m

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSAppCenterUserDefaults.h"
+#import "MSAppCenterInternal.h"
+#import "MSLogger.h"
+
+static NSString *const kMSAppCenterUserDefaultsMigratedKey = @"310UserDefaultsMigratedKey";
+
+static MSAppCenterUserDefaults *sharedInstance = nil;
+static dispatch_once_t onceToken;
+
+// MSAppCenterUserDefaults keys to be migrated.
+static NSMutableDictionary<NSString *, NSString *> *keysToMigrate;
+
+@implementation MSAppCenterUserDefaults
+
++ (void)load {
+  keysToMigrate = [NSMutableDictionary new];
+}
+
++ (void)addKeysToMigrate:(NSDictionary<NSString *, NSString *> *)keys {
+  [keysToMigrate addEntriesFromDictionary:keys];
+}
+
++ (instancetype)shared {
+  dispatch_once(&onceToken, ^{
+    sharedInstance = [[MSAppCenterUserDefaults alloc] init];
+    NSDictionary *changedKeys = @{
+      @"MSChannelStartTimer" : @"MSAppCenterChannelStartTimer",         // MSChannelUnitDefault
+      @"pastDevicesKey" : @"MSAppCenterPastDevicesKey",                 // MSDeviceTrackerPrivate
```

Comment:
Sometimes we use the `Key` suffix like for this key, no big deal but this is our chance for harmonizing so let's remove all the `key` suffix from the keys string I think it's useless.